ios/brave-ios/Sources/Brave/Frontend/Settings/General/SearchEngines/Custom Engines/CustomEngineViewController.swift

```diff
@@ -130,43 +166,27 @@ class CustomEngineViewController: UIViewController {
     tableView.do {
       $0.register(CustomEngineURLInputTableViewCell.self)
       $0.register(CustomEngineTitleInputTableViewCell.self)
+      $0.register(CustomEngineAddButtonCell.self)
       $0.registerHeaderFooter(SearchEngineTableViewHeader.self)
       $0.dataSource = self
       $0.delegate = self
     }
   }
 
   private func doLayout() {
+    view.backgroundColor =
+      traitCollection.userInterfaceStyle == .dark
+      ? .secondarySystemGroupedBackground : .systemGroupedBackground
```

Comment:
If you're going to use `traitCollection` for anything you need to make sure to re-evaluate when it changes via `traitCollectionDidChange`